### PR TITLE
Introduce the setting "engine-replica-timeout" for changing iSCSI timeout value

### PR DESCRIPTION
**Pull request by derekbit**
https://github.com/longhorn/longhorn/issues/4491

Depends on https://github.com/longhorn/longhorn-engine/pull/762

Signed-off-by: Derek Su [derek.su@suse.com](mailto:derek.su@suse.com)

### CODE REVIEW

**Comment by innobead** [comment]
@yangchiu could we allow members able to restart the failed PR check?

**Comment by yangchiu** [comment]
https://github.com/drone/drone-admit-members/blob/master/plugin/plugin.go#L64

We can create a team and put all members into this team to achieve this automatically, currently we don't have such team https://github.com/orgs/longhorn/teams.

But I've manually modified @derekbit 's role to admin. @derekbit can check if it works, then probably I can manually modify all member's role to admin too.

**Comment by innobead** [comment]
> 



> https://github.com/drone/drone-admit-members/blob/master/plugin/plugin.go#L64
> 
> We can create a team and put all members into this team to achieve this automatically, currently we don't have such team https://github.com/orgs/longhorn/teams.
> 
> But I've manually modified @derekbit 's role to admin. @derekbit can check if it works, then probably I can manually modify all member's role to admin too.
> 
> Also I think `git push -f ` can retrigger the PR check?

Use this team, https://github.com/orgs/longhorn/teams/dev.

**Comment by derekbit** [comment]
> https://github.com/drone/drone-admit-members/blob/master/plugin/plugin.go#L64
> 
> We can create a team and put all members into this team to achieve this automatically, currently we don't have such team https://github.com/orgs/longhorn/teams.
> 
> But I've manually modified @derekbit 's role to admin. @derekbit can check if it works, then probably I can manually modify all member's role to admin too.
> 
> Also I think `git push -f ` can retrigger the PR check?

Works. Thank you.

**Comment by yangchiu** [comment]
> > 
> 
> > https://github.com/drone/drone-admit-members/blob/master/plugin/plugin.go#L64
> > We can create a team and put all members into this team to achieve this automatically, currently we don't have such team https://github.com/orgs/longhorn/teams.
> > But I've manually modified @derekbit 's role to admin. @derekbit can check if it works, then probably I can manually modify all member's role to admin too.
> > Also I think `git push -f ` can retrigger the PR check?
> 
> Use this team, https://github.com/orgs/longhorn/teams/dev.

Currently the team is set to `infra`, if change to `dev`, `infra` members (and me) will lose admin access, and `qa` members will be still unable to restart it. Shall we do the change?

**Comment by innobead** [comment]
> > > 
> > 
> > 
> > > https://github.com/drone/drone-admit-members/blob/master/plugin/plugin.go#L64
> > > We can create a team and put all members into this team to achieve this automatically, currently we don't have such team https://github.com/orgs/longhorn/teams.
> > > But I've manually modified @derekbit 's role to admin. @derekbit can check if it works, then probably I can manually modify all member's role to admin too.
> > > Also I think `git push -f ` can retrigger the PR check?
> > 
> > 
> > Use this team, https://github.com/orgs/longhorn/teams/dev.
> 
> Currently the team is set to `infra`, if change to `dev`, `infra` members (and me) will lose admin access, and `qa` members will be still unable to restart it. Shall we do the change?

Wait. This is not what I want, let me create another ticket, then let's follow up there.

**Comment by derekbit** [comment]
Will update the dependencies if https://github.com/longhorn/longhorn-engine/pull/762 is merged.

File: controller/engine_controller.go [hunk 1]
```diff
@@ -1668,13 +1667,12 @@ func (ec *EngineController) UpgradeEngineProcess(e *longhorn.Engine) error {
 	}
 	defer c.Close()
 
-	sizeRequested, err := ec.ds.IsEngineImageCLIAPIVersionEqualToOrLargerThan(e.Spec.EngineImage, 6)
+	engineCLIAPIVersion, err := ec.ds.GetEngineImageCLIAPIVersion(e.Spec.EngineImage)
```
Comment (innobead) [inline controller/engine_controller.go:1680]: `IsEngineImageCLIAPIVersionEqualToOrLargerThan` seems not used anymore, so suggest removing.

File: types/setting.go [hunk 2]
```diff
@@ -858,6 +861,16 @@ var (
 		ReadOnly: false,
 		Default:  "false",
 	}
+
+	SettingDefinitionEngineReplicaTimeout = SettingDefinition{
```
Comment (innobead) [inline types/setting.go:865]: Remember updating default-setting yaml, and manifests in longhorn/longhorn.

**Review by innobead — CHANGES_REQUESTED** [review]
LGTM, just minor changes needed

File: controller/engine_controller.go (hunk 1 above)
Comment (derekbit) [reply controller/engine_controller.go:1680]: Removed. Thank you.

File: types/setting.go (hunk 2 above)
Comment (derekbit) [reply types/setting.go:865]: Updated. Thank you.

**Comment by derekbit** [comment]
Will update the dependencies after https://github.com/longhorn/longhorn-instance-manager/pull/139 is getting merged.

File: deploy/install/01-prerequisite/04-default-setting.yaml [hunk 3]
```diff
@@ -46,4 +46,6 @@ data:
     #storage-network:
     #failed-backup-ttl:
     #recurring-successful-jobs-history-limit:
-    #recurring-failed-jobs-history-limit:
\ No newline at end of file
+    #recurring-failed-jobs-history-limit:
+    #deleting-confirmation-flag:
```
Comment (innobead) [inline deploy/install/01-prerequisite/04-default-setting.yaml:50]: This was missed before. cc @PhanLe1010

**Review by innobead — APPROVED** [review]

**Pull request merged**